apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt

```diff
@@ -89,10 +91,14 @@ internal class DefaultApolloStore(
       data: D,
       customScalarAdapters: CustomScalarAdapters,
   ): Map<String, Record> {
+    // Capture a local reference so as not to freeze "this"
+    val metadataGenerator = metadataGenerator
```

Comment:
Something's off here. There's no lambda capturing `metadataGenerator`. Also if there were, the same logic would apply to `cacheKeyGenerator`